modules/portal/app/models/DnsChangeNotices.scala

```
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import com.typesafe.config.{Config, ConfigException}
+import models.DnsChangeNoticeType.DnsChangeNoticeType
+import models.DnsChangeStatus.DnsChangeStatus
+import play.api.libs.json.{JsValue, Json}
+import play.api.ConfigLoader
+import net.ceedubs.ficus.Ficus._
+import net.ceedubs.ficus.readers.EnumerationReader._
+
+import scala.collection.JavaConverters._
+
+case class DnsChangeNotices(notices: JsValue)
+
+object DnsChangeNotices {
+  implicit val configLoader: ConfigLoader[DnsChangeNotices] =
+    new ConfigLoader[DnsChangeNotices] {
+      def load(config: Config, path: String): DnsChangeNotices = {
+        val notices = config.getConfigList(path).asScala.map { noticeConfig =>
+          formatDnsChangeNotice(noticeConfig)
+        }
+        DnsChangeNotices(noticesToJson(notices.toList))
+      }
+    }
+
+  def noticesToJson(notices: List[DnsChangeNotice]): JsValue =
+    Json.toJson(
+      notices.map { n =>
+        Map(
+          "status" -> n.status.toString,
+          "alertType" -> n.alertType.toString,
+          "text" -> n.text,
+          "href" -> n.href.getOrElse(""),
+          "hrefText" -> n.hrefText.getOrElse("")
+        )
+      }
+    )
+
+  def formatDnsChangeNotice(config: Config): DnsChangeNotice = {
+    val status = try {
```

in general, the rule is that if there is bad config to fail on load.   Fail fast instead of try/catching here

---

I had to do a try/catch for implementation of ficus. It doesn't allow a fallback if the value given doesn't match the enumeration. For status and alertType i want to accept values outside of the enumeration in the event the supported values change I don't want to break someone's application or if they want to set up their config in anticipation of new values. 

I could go back to the mapping in the earlier commit if we don't like the try and catch handling needed with ficus.

---

@britneywright you mean ficus doesn't throw if it fails?  I.e. if you do `config.as[DnsChangeStatus]("foo")` is just says "ok"?